pyproject.toml

```diff
@@ -60,6 +60,18 @@ pep8-naming = "^0.13.2"
 pyright = "^1.1.275"
 typing-extensions = "^4.4.0"
 
+[tool.poetry.group.docs]
+optional = true
+
+[tool.poetry.group.docs.dependencies]
+sphinx = "^5.3.0"
+sphinx-autodoc-typehints = "^1.19.0"
+furo = "*"
```

Comment:
Why is furo a wildcard dependency here?

Reply:
There are multiple reasons

1. `furo` uses date versioning instead of semantic.
2. `furo` hasn't released any breaking changes, except drops for old Sphinx versions. See [changelog](https://pradyunsg.me/furo/changelog/).
3. `furo` is a pretty old and stable package. GitHub shows 6.8k usages.
4. `furo` updates are usually small, and do not have any really meaningful changes. Only some CSS and HTML changes, but usually nothing more.
5. And the most important, I'm tired of merging dependabot's PR almost every day once in a while.

P.S. I used wildcard here for a year, and I do not remember any situation when `furo` broke something.

Reply:
> And the most important, I'm tired of merging dependabot's PR almost every day once in a while.

Dependabot will still generate PRs that update the lock file, that PR just won't include an update to pyproject.toml. 

The pyproject.toml is more about what dependencies can be used that still result in a runnable version of our package. So if furo is listed as wildcard, anyone with any version of furo installed will not have that version updated (unless they also use our lock file, with the exact version).

For this reason, I like to specify a minimum version for dependencies, to make sure that no one will end up using some really old version, so, even with furo, I'd image the first ever released version probably wouldn't be compatible with the current one anymore.

However since furo doesn't use semvers, instead of using the usual caret requirement syntax, we could use the inequality one, like say `furo>=2022-08-16`, to at least have that minimum version.

Reply:
> Dependabot will still generate PRs that update the lock file, that PR just won't include an update to pyproject.toml.

Oh, yes. I just ignored furo for dependabot and updated it with `poetry update`.

> However since furo doesn't use semvers, instead of using the usual caret requirement syntax, we could use the inequality one, like say `furo>=2022-08-16`, to at least have that minimum version.

Should I open a PR, or you will just commit the change in this branch?

Reply:
Also, there is no way to install dependency without a lock file, `rm poetry.lock && poetry install` will create a new `poetry.lock`. Even `pip install .` can't do it, because `pip` can't read poetry 1.2 groups.

Reply:
This is a bit more relevant and easier to explain for actual release dependencies, which are what's required when this package is installed by others. If we specify a dependency like `dnspython` with say `^1.2.0`, and someone installing already has dnspython in their venv, but their version is at `1.1.0`, it will be updated (unless some dependency requires `<=1.1.0`, or `~1.1.0`, in which case our package would fail installation, since it's dependencies can't be satisfied.

With development dependencies, this isn't that huge of an issue, since it's unlikely that someone working on mcstatus would be installing into an already established venv without a lock file, but that's the case this guards against. Basically, you can create a python venv with any tool you like (for example you can run `python -m venv .venv`). You could then install some really old version of `furo` into this environment through pip (`./.venv/bin/pip install furo==2018.01.25`). Now, you could copy over our `pyproject.toml`, and run `poetry install --with docs` into this environment. At that point, since you didn't copy the lock file and only have the pyproject, poetry will install all of the unsatisfied dependencies, but it will not install furo, since that's already satisfied for the wildcard `*` specification. During this process, poetry will create a lock file for you, with all of the dependency versions listed exactly.

Obviously, nobody will probably do this (though people do sometimes do some weird stuff, so I wouldn't even be that surprised), nevertheless, it shows exactly what the issue with this kind of specification is. And we can't really know how everyone's poetry is configured, because while the above case isn't very likely, what might be something that people do is have poetry configured to use a single environment, perhaps even the global one, and just install all of the packages there. 

This certainly isn't recommended and will likely end up causing some dependencies to not be installable since they require a newever version of another dep, which some existing package isn't compatible with. This is why we use virtual environments. Still though, if someone would do this, and from whatever other project, this environment already contained `furo` at some older version, our project would be "installable", even if this furo version is really old and perhaps not compatible.

Reply:
> Should I open a PR, or you will just commit the change in this branch?

Fixed in https://github.com/py-mine/mcstatus/commit/799d72cb8a7f0e21d40004782d77a271ae4db161

Reply:
You forgot to run `poetry lock` after the change. I got this after `poetry lock --no-cache --no-update`:

```patch
diff --git a/poetry.lock b/poetry.lock
index a0cc769..14cd04e 100644
--- a/poetry.lock
+++ b/poetry.lock
@@ -1946,4 +1946,4 @@ testing = ["flake8 (<5)", "func-timeout", "jaraco.functools", "jaraco.itertools"
 [metadata]
 lock-version = "2.0"
 python-versions = ">=3.8,<4"
-content-hash = "4602ca655e172d8a22c07ad0432c6c7e0cbae158eccf2e645b57faa3b834e51e"
+content-hash = "8ad23d6329ce4df6ad136c47d22f2d4951e1fbfd6b612306bcd82eec0bd2cfdd"
```

Reply:
Ah, that happened after merge not after this fix, but sure, I can update the lock file